File: 03-O-Número-é-Palíndromo/solucao.js
function ehPalindromo(numero){
    const numeroOriginal = numero;
    let invertido = 0;

    while(numero > 0){
        let digito = numero % 10;
        invertido = (invertido * 10) + digito;
        numero = Math.floor(numero / 10);
    }

    return numeroOriginal === invertido;
}

//Casos de teste
console.log(ehPalindromo(4554));
console.log(ehPalindromo(123));

File: 03-O-Número-é-Palíndromo/solucao.py
def eh_palindromo(numero:int) -> bool:
    numero_original = numero
    invertido = 0

    while numero > 0:
        digito = numero % 10
        invertido = (invertido * 10) + digito
        numero = numero // 10

    return numero_original == invertido

## Casos de teste
if __name__ == "__main__":
    print(f"121 é palíndromo? {eh_palindromo(121)}")
    print(f"4554 é palíndromo? {eh_palindromo(4554)}")
    print(f"123 é palíndromo? {eh_palindromo(123)}")
    print(f"7 é palíndromo? {eh_palindromo(7)}")
